x/finality/keeper/gov.go

```diff
@@ -113,9 +120,38 @@ func (k Keeper) HandleResumeFinalityProposal(ctx sdk.Context, fpPksHex []string,
 
 		distCache.ApplyActiveFinalityProviders(params.MaxActiveFinalityProviders)
 
+		// add active fps to the all active fps set if it does not exist
+		for pk, fp := range distCache.GetActiveFinalityProviderSet() {
+			allActiveFps[pk] = fp
+		}
+
 		// set the voting power distribution cache of the current height
 		k.SetVotingPowerDistCache(ctx, h, distCache)
 	}
 
+	// it is possible that some inactive fps become active after the proposal
+	// therefore, we need to ensure every active finality provider has signing info
+	for pk, dc := range allActiveFps {
+		hasSigningInfo, err := k.FinalityProviderSigningTracker.Has(ctx, dc.BtcPk.MustMarshal())
+		if err != nil {
+			return fmt.Errorf("failed to get signing info from tracker for fp %s - %w", pk, err)
+		}
+
+		if hasSigningInfo {
+			continue
+		}
+
+		signingInfo := ftypes.NewFinalityProviderSigningInfo(
```

Comment:
adding signing info is not only think that happens after fp is activated. Other other thing that happens is emitting events - https://github.com/babylonlabs-io/babylon/blob/db20d092d8fd1f3a23133d7d90341ad6b433777f/x/finality/keeper/power_dist_change.go#L97.

Do you think we should do it here to avoid breaking anybody depending on events  ? @gitferry @RafilxTenfen 

Also maybe we can double check if there aren't other missing things which happens with each fp actiavation / deactivation in the normal path.

Reply:
good catch, we should emit events

Reply:
done at https://github.com/babylonlabs-io/babylon/pull/860/commits/a5ef731ca79a45b9a4795e6abda9d86c2448d320